src/test/kotlin/dev/stalla/model/factory/ItunesCategoryFactoryTest.kt

```diff
@@ -0,0 +1,148 @@
+    )
+
+    private val validNestedItunesCategoryNames = listOf(
+        "Books",
+        "Design",
+        "Fashion & Beauty",
+        "Food",
+        "Performing Arts",
+        "Visual Arts",
+        "Careers",
+        "Entrepreneurship",
+        "Investing",
+        "Management",
+        "Marketing",
+        "Non-Profit",
+        "Comedy Interviews",
+        "Improv",
+        "Stand-Up",
+        "Courses",
+        "How To",
+        "Language Learning",
+        "Self-Improvement",
+        "Comedy Fiction",
+        "Drama",
+        "Science Fiction",
+        "Alternative Health",
+        "Fitness",
+        "Medicine",
+        "Mental Health",
+        "Nutrition",
+        "Sexuality",
+        "Education for Kids",
+        "Parenting",
+        "Pets & Animals",
+        "Stories for Kids",
+        "Animation & Manga",
+        "Automotive",
+        "Aviation",
+        "Crafts",
+        "Games",
+        "Hobbies",
+        "Home & Garden",
+        "Video Games",
+        "Music Commentary",
+        "Music History",
+        "Music Interviews",
+        "Business News",
+        "Daily News",
+        "Entertainment News",
+        "News Commentary",
+        "Politics",
+        "Sports News",
+        "Tech News",
+        "Buddhism",
+        "Christianity",
+        "Hinduism",
+        "Islam",
+        "Judaism",
+        "Religion",
+        "Spirituality",
+        "Astronomy",
+        "Chemistry",
+        "Earth Sciences",
+        "Life Sciences",
+        "Mathematics",
+        "Natural Sciences",
+        "Nature",
+        "Physics",
+        "Social Sciences",
+        "Documentary",
+        "Personal Journals",
+        "Philosophy",
+        "Places & Travel",
+        "Relationships",
+        "Baseball",
+        "Basketball",
+        "Cricket",
+        "Fantasy Sports",
+        "Football",
+        "Golf",
+        "Hockey",
+        "Rugby",
+        "Running",
+        "Soccer",
+        "Swimming",
+        "Tennis",
+        "Volleyball",
+        "Wilderness",
+        "Wrestling",
+        "After Shows",
+        "Film History",
+        "Film Interviews",
+        "Film Reviews",
+        "TV Reviews"
+    )
+
+    @Test
+    fun `should retrieve all defined iTunes categories from the interface factory method`() {
+        for (simpleCategory in validSimpleItunesCategoryNames) {
+            assertThat(ItunesCategory.from(simpleCategory)).isNotNull().all {
+                prop(ItunesCategory::name).isEqualTo(simpleCategory)
+            }
+        }
+        for (nestedCategory in validNestedItunesCategoryNames) {
+            assertThat(ItunesCategory.from(nestedCategory)).isNotNull().all {
+                prop(ItunesCategory::name).isEqualTo(nestedCategory)
+            }
+        }
+    }
+
+    @Test
+    fun `should not retrieve an undefined iTunes category from the interface factory method`() {
+        assertThat(ItunesCategory.from("itunes category")).isNull()
+    }
```

Comment:
Would add a simple test here enumerating all the vals on the `Factory` companion, and comparing them with the above lists, making sure they're identical — this way we are sure we haven't forgotten anything